internal/app/orchestrator/downstream.go

```diff
@@ -78,3 +80,17 @@ func (d *downstreamResponseMap) deleteAll(watchers map[*gcp.Request]bool) {
 		}
 	}
 }
+
+func (d *downstreamResponseMap) keys(m *mapper.Mapper) ([]string, error) {
+	d.mu.RLock()
+	defer d.mu.RUnlock()
+	var keys []string
```

Comment:
Do we need to dedupe since 2 requests can map to the same keys?

Reply:
yep, good catch.